exp/runtime/internal/controllers/extensionconfig_controller.go

```diff
@@ -224,12 +227,16 @@ func discoverExtensionConfig(ctx context.Context, runtimeClient runtimeclient.Cl
 // Note: This was implemented to behave similar to the cert-manager cainjector.
 // We couldn't use the cert-manager cainjector because it doesn't work with CustomResources.
 func reconcileCABundle(ctx context.Context, client client.Client, config *runtimev1.ExtensionConfig) error {
+	log := ctrl.LoggerFrom(ctx)
+
 	secretNameRaw, ok := config.Annotations[runtimev1.InjectCAFromSecretAnnotation]
 	if !ok {
 		return nil
 	}
 	secretName := splitNamespacedName(secretNameRaw)
 
+	log.Info(fmt.Sprintf("Injecting CA Bundle into extension '%s/%s' from secret '%s'", config.Namespace, config.Name, secretName))
```

Comment:
`reconcileCABundle` is also called during the warmup. During wram up this will be called for multiple ExtensionConfigs and since that is not a reconciler the logger wont have the k/v pairs attached, hence adding it to the log here.

Once way might be to add the corrsponding k/v pairs to the logger in the ctx when this function is being called in the warm up. Does that sounds good?

Reply:
Looks like you agree. (https://github.com/kubernetes-sigs/cluster-api/pull/6832#discussion_r914440742) . Will make the change.